client_test.go

```diff
@@ -459,12 +460,7 @@ func Test_Client(t *testing.T) {
 		default:
 		}
 
-		stopCtx, stopCancel := context.WithTimeout(ctx, 5*time.Second)
-		t.Cleanup(stopCancel)
```

Comment:
I took this out because with the addition of `require.ErrorIs` above, it revealed that it occasionally causes an intermittent failure as it kicks in and supersedes the shutdown's cancellation. I don't think this is needed anyway because if shutdown's context cancellation doesn't succeed, I don't think we could expect this one to succeed either. Also, not having keeps the test more succinct.